### PR TITLE
Build docker image for preview

### PR DESCRIPTION
**Pull request by schmir-at-bb**
Add a preview NODE_ENV and build a nginx based docker image. Upload that to dockerhub.

Build the image with:

  yarn docker-preview

and start a local webserver with:

  docker run -p 8080:80 --rm -it preview.trustlines.foundation

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -56,3 +83,8 @@ workflows:
       - build:
           requires:
             - install
+      - docker:
+          context: docker-credentials
+          requires:
+            - build
```
Comment (weilbith) [inline .circleci/config.yml]: I think to require `install` should be enough. The `docker` job does also build the project, but slightly different. Their shared requirement should be the installation part.

**Comment by schmir-at-bb** [comment]
> I'm not too happy that we have a fourth environment file now. Actually I thought we could use the development version. But I see your point. man_shrugging

I think the example environment should be removed eventually. At least it's badly named, because the way we use it, it should be called default environment.

> How is your plan to use that image now? Isn't there a filter for a specific branch missing? thinking

I would probably use this together with watchtower. Building and uploading for every branch was done on purpose. That way should be able to easily have a preview for multiple branches (if we feel the need to do that).  And we do have an archive of everything.

File: .circleci/config.yml (hunk 1 above)
Comment (schmir-at-bb) [reply .circleci/config.yml]: changed, thanks.

**Comment by weilbith** [comment]
> I would probably use this together with watchtower. Building and uploading for every branch was done on purpose. That way should be able to easily have a preview for multiple branches (if we feel the need to do that). And we do have an archive of everything.

Okay that matches exactly what I thought of.
So let's consider there are multiple PRs open. And Andreas still has to review the first one whiel the second needs some continue work due to a technical review or what ever. How can I no tell the serving preview machine to pull one of the old image builds again? I'm not even sure if I could do that manually without watchtower to update the container again... :thinking:

**Pull request merged**